presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -582,6 +591,16 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
     public void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToRevoke = privilegeGrantInfoSet.stream()
+                    .filter(requestedPrivilege -> existingPrivileges.containsAll(parsePrivilege(requestedPrivilege)))
```

Comment:
Oh, that will be taken care of. The name `containsAll` is a bit misleading here, since we only check `contains` actually. The reason I needed to call `containsAll` instead of `contains` is because `parsePrivilege(requestedPrivilege)` returns a `Set` even for a single privilege. Also, requestedPrivilege would never be a set like ("SELECT","INSERT"). `requestedPrivilege` would take values one by one from the set (java streams API call).